packages/web/src/components/createCell.tsx

```diff
@@ -273,15 +273,26 @@ export function createCell<
     const options = beforeQuery(variables as CellProps)
     const query = typeof QUERY === 'function' ? QUERY(options) : QUERY
 
-    // queryRest includes `variables: { ... }`, with any variables returned
-    // from beforeQuery
-    let {
-      // eslint-disable-next-line prefer-const
-      error,
-      loading,
-      data,
-      ...queryResult
-    } = useQuery(query, options)
+    const useQueryHook = RWJS_ENV.RWJS_EXP_STREAMING_SSR
+      ? useSuspenseQuery
+      : useQuery
+
+    let error = null
+    let loading = false
+    let data = null
+    let queryResult = null
+
+    const hookOutput = useQueryHook(query, options)
+
+    // This terrible stuff is here, because useSuspenseQuery
+    // doesnt give us loading or error. It's handled by the Suspense fallback and ErrorBoundary
+    // And you can't conditionally call a hook inside the if statement :/
```

Comment:
In the workaround I did, I swapped out in the temporary suspense client:

```ts
  if (RWJS_ENV.RWJS_EXP_STREAMING_SSR) {
    useQueryHookSwap = useSuspenseQuery as unknown as typeof useQuery
  }
```

and that seems sort of what you are doing here: swapping out in the cell.

Which makes sense for now.

And then by adding the new hook to the provider, types are cleaner and there is no mystery about which hook is being uses: suspense or classic query.

Seem like a reasonable solution at moment,.

Reply:
Yeah either way is fine by me :) 

I think we both need Apollo 3.8.x merged in soon haha.